### PR TITLE
Added support for fractional font sizes.

### PR DESCRIPTION
**Pull request by phao**

### CODE REVIEW

File: src/editor.c [hunk 1]
```diff
@@ -4460,12 +4460,12 @@ void editor_set_font(GeanyEditor *editor, const gchar *font)
 	g_return_if_fail(editor);
 
 	pfd = pango_font_description_from_string(font);
-	size = pango_font_description_get_size(pfd) / PANGO_SCALE;
+	size = (100*pango_font_description_get_size(pfd)) / PANGO_SCALE;
```
Comment (b4n) [inline src/editor.c]: why multiply by 100 and divide later?  wouldn't it be simpler and everything to just make `size` a `gdouble` and get a real straight away?

File: src/sciwrappers.c [hunk 2]
```diff
@@ -856,6 +856,18 @@ gint sci_find_text(ScintillaObject *sci, gint flags, struct Sci_TextToFind *ttf)
...
+/** Sets the font for a particular style.
+ * @param sci Scintilla widget.
+ * @param style The style.
+ * @param font The font name.
+ * @param size The font (fractional) size. */
+void sci_set_font_fract(ScintillaObject *sci, gint style, const gchar *font, double size)
+{
+	SSM(sci, SCI_STYLESETFONT, (uptr_t) style, (sptr_t) font);
+
+  // Adding 0.5 is to round.
```
Comment (b4n) [inline src/sciwrappers.c]: indentation should be 1tab, not 2 spaces

File: src/sciwrappers.c [hunk 3]
```diff
@@ -864,8 +876,7 @@ gint sci_find_text(ScintillaObject *sci, gint flags, struct Sci_TextToFind *ttf)
  * @param size The font size. */
 void sci_set_font(ScintillaObject *sci, gint style, const gchar *font, gint size)
 {
-	SSM(sci, SCI_STYLESETFONT, (uptr_t) style, (sptr_t) font);
-	SSM(sci, SCI_STYLESETSIZE, (uptr_t) style, size);
+  sci_set_font_fract(sci, style, font, size);
```
Comment (b4n) [inline src/sciwrappers.c]: ditto

File: src/sciwrappers.c [hunk 4]
```diff
@@ -856,6 +856,18 @@ gint sci_find_text(ScintillaObject *sci, gint flags, struct Sci_TextToFind *ttf)
 	return (gint) SSM(sci, SCI_FINDTEXT, (uptr_t) flags, (sptr_t) ttf);
 }
 
+/** Sets the font for a particular style.
+ * @param sci Scintilla widget.
+ * @param style The style.
+ * @param font The font name.
+ * @param size The font (fractional) size. */
+void sci_set_font_fract(ScintillaObject *sci, gint style, const gchar *font, double size)
```
Comment (b4n) [inline src/sciwrappers.c]: we use GLib's type aliases, so `gdouble` (though it doesn't change anything here, that's only style)

File: src/sciwrappers.h [hunk 5]
```diff
@@ -72,6 +72,7 @@ gchar				sci_get_char_at				(ScintillaObject *sci, gint pos);
 
 void				sci_scroll_caret			(ScintillaObject *sci);
 gint				sci_find_text				(ScintillaObject *sci, gint flags, struct Sci_TextToFind *ttf);
+void				sci_set_font_fract				(ScintillaObject *sci, gint style, const gchar *font, double size);
```
Comment (b4n) [inline src/sciwrappers.h]: there is one tab too many on the alignment on the right, we use tabs of width 4 (see HACKING file on the repository)

**Comment by elextr** [comment]
@codebrainz are you sure that the @b4n --excessively-pedantic options set won't complain about passing ints to doubles if we do that :)

Otherwise @phao just put a space between the two stars `/* *` for now.

**Comment by elextr** [comment]
@b4n ping, fixed to your satisfaction?

**Comment by b4n** [comment]
See #2250

**Pull request closed without merging**